akka-testkit-typed/src/main/scala/akka/typed/testkit/BehaviourTestkit.scala

```diff
@@ -126,3 +95,80 @@ class EffectfulActorContext[T](_name: String, _initialBehavior: Behavior[T], _ma
     super.schedule(delay, target, msg)
   }
 }
+
+@ApiMayChange
+object BehaviorTestkit {
+  def apply[T](initialBehavior: Behavior[T], name: String): BehaviorTestkit[T] =
+    new BehaviorTestkit[T](name, initialBehavior)
+  def apply[T](initialBehavior: Behavior[T]): BehaviorTestkit[T] =
+    apply(initialBehavior, "ctx")
+}
+
+/**
+ * Used for testing [[Behavior]]s. Stores all effects e.g. Spawning of children,
+ * watching and offers access to what effects have taken place.
+ */
+@ApiMayChange
+class BehaviorTestkit[T](_name: String, _initialBehavior: Behavior[T]) {
+  import Effect._
+
+  // really this should be private, make so when we port out tests that need it
+  private[akka] val ctx = new EffectfulActorContext[T](_name)
+
+  /**
+   * Requests the oldest [[Effect]] or [[NoEffects]] if no effects
+   * have taken place
+   */
+  def requestEffect(): Effect = ctx.effectQueue.poll() match {
```

Comment:
is `retrieve` better than `request`?

Reply:
Given the synchronous context I think retrieve is better